### ISSUE TITLE
Feature request: Support for systemd.volatile=overlay with nfsroot

### ISSUE THREAD

**Issue by 0n0w1c** [archlinux/mkinitcpio#68]
dracut currently offers this support but I would prefer to stay with mkinitcpio. systemd.volatile=overlay is superior to other methods of implementing an overlayfs at boot time as upper and lower mounts are visible.

**Comment by grazzolini** [comment]
I have the same issue as the previous mkinitcpio maintainer. I have no NFS setup to test this. If you could test this: https://bugs.archlinux.org/task/28287#comment104020 it would be great.

**Comment by 0n0w1c** [comment]
Thank you, yes, I will be happy to.

**Comment by grazzolini** [comment]
Yes, precisely.

**Comment by 0n0w1c** [comment]
Yes, in a nutshell, if you make the above changes and properly configure the kernel arguments for ip= and nfsroot= a network mounted root filesystem via nfs4 does work.

For example: ip=:::::eth0:dhcp nfsroot=10.1.1.9:/srv/nfs/root

One should make the changes for nfs4 more sustainable, by following the relevant instructions [here](https://wiki.archlinux.org/title/Diskless_system).

**Comment by grazzolini** [comment]
What changes do you mean, specifically?

**Comment by 0n0w1c** [comment]
Install mkinitcpio-nfs-utils

mkinitcpio.conf
```
- MODULES=()
+ MODULES=(sunrpc)

- BINARIES=()
+ BINARIES=(/usr/bin/mount.nfs4)

- HOOKS=(base udev autodetect modconf block filesystems keyboard fsck)
+ HOOKS=(base udev autodetect modconf block filesystems keyboard fsck net)
```

/usr/lib/initcpio/hooks/net
```
- nfsmount ${nfs_option:+-o ${nfs_option}} "${nfs_server}:${nfs_path}" "$1"
+ mount.nfs4 ${nfs_option:+-o ${nfs_option}} "${nfs_server}:${nfs_path}" "$1"
```
* Note this is not the best way to do this, as this file can be overwritten by a package update.

Modify the kernel parameters to include something like:
nfsroot=10.208.101.11:/srv/nfs/root ip=dhcp

Remake the initramfs and reboot.

I think that is all that was required.

**Comment by 0n0w1c** [comment]
It seems that currently there is an issue with `systemd.volatile=overlay` and dracut. You can see my tickets [here](https://github.com/dracutdevs/dracut/issues/1738) and [here](https://github.com/systemd/systemd/issues/22645) along with my hacky "fix".

But it covers the issue and should help with what mkinitcpio would need to support `systemd.volatile=overlay`.

**Comment by 0n0w1c** [comment]
I have been attempting to get this to work. Currently stuck on systemd-networkd failing with status of 217/USER with `User=systemd-network`. Not exactly sure what I am missing to get "Users="  to work in initramfs... if anyone has a hint, it would be appreciated.
